indexing-service/src/main/java/org/apache/druid/indexing/overlord/setup/WorkerTierSpec.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.indexing.overlord.setup;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.util.Map;
+import java.util.Objects;
+
+public class WorkerTierSpec
+{
+  // key: taskType, value: tierConfig
+  private final Map<String, TierConfig> tierMap;
+  private final boolean strong;
+
+  @JsonCreator
+  public WorkerTierSpec(
+      @JsonProperty("tierMap") Map<String, TierConfig> tierMap,
+      @JsonProperty("strong") boolean strong
+  )
+  {
+    this.tierMap = tierMap;
+    this.strong = strong;
+  }
+
+  @JsonProperty
+  public Map<String, TierConfig> getTierMap()
+  {
+    return tierMap;
+  }
+
+  @JsonProperty
+  public boolean isStrong()
+  {
+    return strong;
+  }
+
+  @Override
+  public boolean equals(final Object o)
+  {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+    final WorkerTierSpec that = (WorkerTierSpec) o;
+    return strong == that.strong &&
+           Objects.equals(tierMap, that.tierMap);
+  }
+
+  @Override
+  public int hashCode()
+  {
+    return Objects.hash(tierMap, strong);
+  }
+
+  @Override
+  public String toString()
+  {
+    return "WorkerTierSpec{" +
+           "tierMap=" + tierMap +
+           ", strong=" + strong +
+           '}';
+  }
+
+  public static class TierConfig
+  {
+    private final String defaultTier;
+    // key: datasource, value: tier
+    private final Map<String, String> tiers;
```

Comment:
How about `datasourceToTier` or `datasource2Tier`?

Reply:
@egor-ryashin ?

Reply:
@QiuMM I think there's no strong argument for choosing between `valueByKey` or `keyToValue` template, I would take the one which is more common throughout the project.

Reply:
instead of supporting only a fixed usecase like tiers per datasource or tiers per indexing type, why not just support a "tier" property within the "context" section of an ingestion spec?

This would allow for arbitrary tier mappings including the ones you have in mind.
The context section is already used for those purposes in the context of queries to set priorities, timeouts etc. I think it would therefore also be a valid design choice to support standard context properties for ingestion tasks.
`
{
  "type": "index_hadoop",
  "context": {
     "tier": "my-very-personal-tier-token", 
     ...
  }
  ...
}


Good point about the autoscaler. its interface should then probably also receive a tier name for which a scale operation should happen...

Reply:
@sascha-coenen thanks for your suggestion. I had also considered this, but it would involve much more code modification which is not worth to do. My solution is undoubtedly logical within current architecture. And it's enough to use I think.